code/modules/lighting/lighting_source.dm

```diff
@@ -1,6 +1,12 @@
 // This is where the fun begins.
 // These are the main datums that emit light.
 
+// Thanks to Lohikar for flinging this tiny bit of code at me, increasing my brain cell count from 1 to 2 in the process.
+// This macro will only offset up to 1 tile, but anything with a greater offset is an outlier and probably should handle its own lighting offsets.
+// Anything pixelshifted 16px or more will be considered on the next tile.
+#define GET_APPROXIMATE_PIXEL_DIR(PX, PY) ((!(PX) ? 0 : ((PX >= 16 ? EAST : (PX <= -16 ? WEST : 0)))) | (!PY ? 0 : (PY >= 16 ? NORTH : (PY <= -16 ? SOUTH : 0))))
+#define UPDATE_APPROXIMATE_PIXEL_TURF var/_mask = GET_APPROXIMATE_PIXEL_DIR(top_atom.pixel_x, top_atom.pixel_y); pixel_turf = _mask ? (get_step(source_turf, _mask) || source_turf) : source_turf
```

Comment:
You could do
```
#define UPDATE_APPROXIMATE_PIXEL_TURF(pixel_turf) do {\
var/_mask = GET_APPROXIMATE_PIXEL_DIR(top_atom.pixel_x, top_atom.pixel_y);\
pixel_turf = _mask ? (get_step(source_turf, _mask) || source_turf) : source_turf } while (0)
```
instead, this lets you not leak the _mask variable. You could also refactor the define so that it takes in source_turf, pixel_turf and either mask (allowing you to get rid of the scope) or top_atom, that way the define is not tied to specifically named proc-scoped variables existing.

Reply:
I'd have to check this, but I think just indenting under if(TRUE) would accomplish the same thing. Hell I'm not even sure that's necessary, I was under the impression just a {} block would work as well.

Reply:
{ } does not mean anything in byond, do { } while (0) is the lummox-approved way of creating a scope, and is optimized out by the compiler.